fmskill/compare.py

```diff
@@ -1282,19 +1282,29 @@ def __init__(self, observation, modeldata):
         if not isinstance(modeldata, list):
             modeldata = [modeldata]
         for j, data in enumerate(modeldata):
-            df = data.to_dataframe()
+            df = self._model2obs_interp(self.observation, data)
+            if (len(df) > 0) and (len(df) == len(self.observation.df)):
+                ok = self._obs_mod_xy_distance_acceptable(df, self.observation.df)
+                df.iloc[~ok, 1] = np.nan
+                if sum(ok) == 0:
+                    warnings.warn("no (spatial) overlap between model and observation points")
             if j == 0:
-                df[self.obs_name] = observation.df.iloc[:, -1]
                 cols = list(df.columns)
-                cols = list((*cols[0:2], *cols[:1:-1]))
+                cols = list((*cols[:1:-1], *cols[0:2]))
                 self.df = df[cols]
             else:
-
                 self.df[self.mod_names[j]] = df[self.mod_names[j]]
 
         self.df = self.df.dropna()
         # TODO: add check
 
+    def _obs_mod_xy_distance_acceptable(self, df_mod, df_obs):
+        mod_xy = df_mod.iloc[:, [-1, -2]].values
```

Comment:
Sorry for sounding like a broken record, but
referring to things (columns) by position is not robust in case of refactoring.

It seems a bit odd that the last two columns are used in one case, and the first two columns in the other.

Reply:
see https://github.com/DHI/fmskill/pull/45/commits/df05f42b896276c2716be5817227d52755a2754d